### ISSUE TITLE
[ek_ra8d1] Image rendering crashed

### ISSUE THREAD

**Issue by kisvegabor** [lvgl/lv_port_renesas_ek-ra8d1_gcc#1]
In LVGL `lv_draw_sw_image` is quite complex and the `lv_draw_dave2d_image` is mostly its duplication. This way it's hard to follow the changes of `lv_draw_sw_image` in `lv_draw_dave2d_image`. 

Probably the reason of the crash is a missing bugfix too.

I'll make a generic way  `lv_draw_sw_image` more generic an reusable.

**Comment by jeremy-baker** [comment]
Yes, there was a bug fix in lv_draw_sw_image, that was missing from lv_draw_dave2d_image, I have put the fix in, and lv_draw_dave2d_image works again.

**Comment by kisvegabor** [comment]
Shall it work with the latest code from the repo too?

**Comment by kisvegabor** [comment]
I've just added the new image draw helpers [here](https://github.com/lvgl/lvgl/tree/feat/dave2d_image), however interestingly it freezes in  [d2_flushframe](https://github.com/lvgl/lvgl/blob/feat/dave2d_image/src/draw/renesas/dave2d/lv_draw_dave2d_image.c#L233).

I kept `img_draw_core` the same so I don't why it freezes. It's just spinning in the idle loop. Do you have any idea?

**Comment by jeremy-baker** [comment]
https://github.com/jeremy-baker/lvgl/commit/bd1224ae3ce66b405f3af5d8d518e7efa7a99a7c
with this commit, it works for me

**Comment by kisvegabor** [comment]
I've tested it and it works on my end too. I added the changes to LVGL.
